hedera-node/hedera-app/src/main/java/com/hedera/node/app/tss/stores/WritableTssStore.java

```diff
@@ -57,44 +71,100 @@ public WritableTssStore(@NonNull final WritableStates states) {
+     */
     public void put(@NonNull final TssVoteMapKey tssVoteMapKey, @NonNull final TssVoteTransactionBody txBody) {
         requireNonNull(tssVoteMapKey);
         requireNonNull(txBody);
         tssVoteState.put(tssVoteMapKey, txBody);
     }
 
+    /**
+     * Persists a new {@link EntityNumber} with given {@link TssEncryptionKeyTransactionBody} into the state.
+     *
+     * @param entityNumber      the corresponding Node ID to the {@link TssEncryptionKeyTransactionBody} to be persisted
+     * @param txBody            the {@link TssEncryptionKeyTransactionBody} for the given node to be persisted
+     */
     public void put(@NonNull final EntityNumber entityNumber, @NonNull final TssEncryptionKeyTransactionBody txBody) {
         requireNonNull(entityNumber);
         requireNonNull(txBody);
         tssEncryptionKeyState.put(entityNumber, txBody);
     }
 
+    /**
+     * Persists a new {@link TssStatus} for the current Tss state.
+     *
+     * @param tssStatus the {@link TssStatus} to be persisted
+     */
     public void put(@NonNull final TssStatus tssStatus) {
         requireNonNull(tssStatus);
         tssStatusState.put(tssStatus);
     }
 
+    /**
+     * Removes a {@link TssMessageTransactionBody} from the state.
+     *
+     * @param tssMessageMapKey for which the {@link TssMessageTransactionBody} to be removed.
+     */
     public void remove(@NonNull final TssMessageMapKey tssMessageMapKey) {
         requireNonNull(tssMessageMapKey);
         tssMessageState.remove(tssMessageMapKey);
     }
 
+    /**
+     * Removes a {@link TssVoteTransactionBody} from the state.
+     *
+     * @param tssVoteMapKey for which the {@link TssVoteTransactionBody} to be removed.
+     */
     public void remove(@NonNull final TssVoteMapKey tssVoteMapKey) {
         requireNonNull(tssVoteMapKey);
         tssVoteState.remove(tssVoteMapKey);
     }
 
+    /**
+     * Removes a {@link TssEncryptionKeyTransactionBody} from the state.
+     *
+     * @param entityNumber the Node for which the {@link TssEncryptionKeyTransactionBody} to be removed.
+     */
     public void remove(@NonNull final EntityNumber entityNumber) {
         requireNonNull(entityNumber);
         tssEncryptionKeyState.remove(entityNumber);
     }
 
+    /**
+     * Removes EntityNumber (Node ID) from the {@link TssEncryptionKeyTransactionBody} map, but only if
+     * the Node ID is present in neither the active roster's and the candidate roster's entries. {@link RosterEntry}
+     *
+     * @param rostersEntriesNodeIds contains the non-duplicate Node IDs of current active and candidate rosters entries
+     */
+    public void removeIfNotPresent(@NonNull final List<EntityNumber> rostersEntriesNodeIds) {
+        requireNonNull(rostersEntriesNodeIds);
+        tssEncryptionKeyState.keys().forEachRemaining(entityNumber -> {
+            if (!rostersEntriesNodeIds.contains(entityNumber)) {
```

Comment:
Should rosterEntriesNodeIds be a `Set<EntityNumber>` instead of a `List<EntityNumber>`?   Line 159 is `O(n)` where `n` is the length of the list.  Granted `n` is small right now.   But the current algorithm requires comparison with all `n` entries to validate the absence of the entry being tested.   With a `Set` the number of comparisons is cut down by using a hashtable on the back end.

Reply:
Makes sense! Updated